include/rsp.inc

```diff
@@ -473,138 +732,229 @@ makeOpInstruction vsubc, 0b010101
+        .error "Invalid element"
+        .exitm
+    .endif
+
+    .ifne ((\offset) % \size)
+        .error "Invalid offset - must be multiple of \size"
+        .exitm
+    .endif
+
+    .ifge (\offset)
+        .ifgt ((\offset) - \rangemax)
+            .error "Invalid offset - valid range: [\rangemin, \rangemax]"
+            .exitm
+        .endif
+
+        \mode\()Vector \opcode, \vt, \element, ((\offset) / \size), \base
+    .else
+        .iflt ((\offset) - \rangemin)
+            .error "Invalid offset - valid range: [\rangemin, \rangemax]"
+            .exitm
+        .endif
+
+        \mode\()Vector \opcode, \vt, \element, (128 + ((\offset) / \size)), \base
+    .endif
+.endm
+
+.macro makeLsInstruction mode, elMode, name, opcode, size, rangemin, rangemax
+    # Overloads:
+    #   op vt, element, offset, base
+    #   op vt, offset, base
+    #   op vt.e, offset, base
+    .macro \name vt, elementOrOffset, offsetOrBase, baseOrEmpty
+        \elMode\()VectorElements
+
+        .ifnb \baseOrEmpty
+            lsInstruction \mode, \opcode, \vt, \elementOrOffset, \offsetOrBase, \baseOrEmpty, \size, \rangemin, \rangemax
+            .exitm
+        .endif
+
+        \elMode\()VectorAccessors
+        encodeVectorRegs
+
+        .iflt (\vt)
+            .error "Invalid element"
+            .exitm
+        .endif
+
+        # Dot-style syntax is encoded as numbers >= 0x200
+        .if (\vt >= 0x200)
+            regVectorAccessors
+            lsInstruction \mode, \opcode, reg.\vt, (\vt & 0xF), \elementOrOffset, \offsetOrBase, \size, \rangemin, \rangemax
+        .else
+            lsInstruction \mode, \opcode, \vt, 0, \elementOrOffset, \offsetOrBase, \size, \rangemin, \rangemax
+        .endif
+    .endm
+.endm
+
+.macro makeLsInstructionQuad mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 16, -1024, 1008
+.endm
+
+.macro makeLsInstructionDouble mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 8, -512, 504
+.endm
+
+.macro makeLsInstructionLong mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 4, -256, 252
+.endm
+
+.macro makeLsInstructionShort mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 2, -128, 126
+.endm
+
+.macro makeLsInstructionByte mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 1, -64, 63
+.endm
+
 /** @brief Load Byte into Vector Register */
-makeLsInstructionByte load, lbv, 0b00000
+makeLsInstructionByte load, byte, lbv, 0b00000
 /** @brief Load Double into Vector Register */
-makeLsInstructionDouble load, ldv, 0b00011
+makeLsInstructionDouble load, byte, ldv, 0b00011
 /** @brief Load Packed Fourth into Vector Register */
-makeLsInstructionQuad load, lfv, 0b01001
+makeLsInstructionQuad load, lane, lfv, 0b01001
 /** @brief Load Packed Half into Vector Register */
-makeLsInstructionQuad load, lhv, 0b01000
+makeLsInstructionQuad load, lane, lhv, 0b01000
```

Comment:
Both lhv and lfv do accept byte offsets.

Reply:
fixed